mibidata/tiff.py

```diff
@@ -88,10 +93,24 @@ def write(filename, image, sed=None, optical=None, ranges=None,
     if not isinstance(image, mi.MibiImage):
         raise ValueError('image must be a mibidata.mibi_image.MibiImage '
                          'instance.')
-    if image.coordinates is None or image.size is None:
-        raise ValueError('Image coordinates and size must not be None.')
-    if image.masses is None or image.targets is None:
-        raise ValueError('Image channels must contain both masses and targets.')
+    missing_required_metadata = []
+    for meta_attr in REQUIRED_METADATA_ATTRIBUTES:
```

Comment:
I think you could do `missing_required_metadata = [m for m in REQUIRED_METADATA_ATTRIBUTES if not getattr(image, m)]`.